scripts/periodic/check_crossref_dois.py

```diff
@@ -0,0 +1,78 @@
+import sys
+import logging
+import requests
+
+from datetime import timedelta
+
+from framework.celery_tasks import app as celery_app
+from website import settings
+from website import mails
+from website.app import init_app
+from django.utils import timezone
+
+import django
+django.setup()
+
+from osf.models import Preprint
+
+from scripts.utils import add_file_logger
+
+logger = logging.getLogger(__name__)
+logging.basicConfig(level=logging.INFO)
+
+
+def check_crossref_dois(dry_run=True):
+    """
+    This script is to check for any DOI confirmation messages we may have missed during downtime and alert admins to any
+    DOIs that have been pending for X number of days.
+    :param dry_run:
+    :return:
+    """
+    # Create one enormous url to check if all our pending crossref DOIs are good, then set all identifers
+    preprints_with_pending_dois = Preprint.objects.filter(preprint_doi_created__isnull=True,
+                                                      is_published=True)
+
+    pending_dois = []
+
+    for preprint in preprints_with_pending_dois:
+        prefix = preprint.provider.doi_prefix
+        pending_dois.append('doi:{}'.format(settings.DOI_FORMAT.format(prefix=prefix, guid=preprint._id)))
+
+    url = '{}/works?filter={}'.format(settings.CROSSREF_JSON_API_URL, ','.join(pending_dois))
+    preprints = requests.get(url).json()['message']['items']
+    for preprint in preprints:
+        guid = preprint['DOI'].split('/')[-1]
+        pending_preprint = preprints_with_pending_dois.get(guids___id=guid)
+        if not dry_run:
+            pending_preprint.set_identifier_values(preprint['DOI'])
+
+def report_stuck_dois(dry_run=True):
+    time_since_published = timedelta(days=settings.DAYS_CROSSREF_DOIS_MUST_BE_STUCK_BEFORE_EMAIL)
+
+    preprints_with_pending_dois = Preprint.objects.filter(preprint_doi_created__isnull=True,
+                                                          is_published=True,
+                                                          date_published__lt=timezone.now() - time_since_published)
+
+    if preprints_with_pending_dois and not dry_run:
+        guids = ', '.join(preprints_with_pending_dois.values_list('guids___id', flat=True))
+        content = 'DOIs for the following preprints have been pending at least {} days: {}'.format(time_since_published.days, guids)
+        mails.send_mail(
+            to_addr=settings.OSF_SUPPORT_EMAIL,
+            mail=mails.CROSSREF_DOIS_PENDING,
+            pending_doi_count=preprints_with_pending_dois.count(),
+            email_content=content,
+        )
+        logger.error('There were {} stuck registrations for CrossRef, email sent to help desk'.format(preprints_with_pending_dois.count()))
+
+
+@celery_app.task(name='scripts.periodic.check_crossref_dois')
+def run_main(dry_run=True):
```

Comment:
It's periodic and all other periodic things that run are written as scripts, so this is more consistent. My understanding is that management commands are always meant to be manually triggered, but I'm unsure,

Reply:
Here's a couple of examples of how I'm handling this in upcoming PRs.

https://github.com/CenterForOpenScience/osf.io/pull/9050/files#diff-c39b88b6fef6c8476d857a5c9da960d6R449

https://github.com/CenterForOpenScience/osf.io/pull/9075/files#diff-ea015ffe216ce38d14dea62852418f34R122

Reply:
The general idea is, just because this will be run periodically, doesn't mean it should only *have* to be run periodically. If crossref is down at 11 and we want to run it again first thing in the morning, having it as a management command makes it easy to do.